### PR TITLE
Add Cloudera repo installation

### PR DESCRIPTION
**Pull request by coreone**
I added the ability to install the Cloudera Yum repo to completely automat the retrieval and installation of zookeeper packages.  I couldn't find a better way to get the GPG key than using an "exec" with curl, but hopefully that isn't too big of an issue.

### CODE REVIEW

File: manifests/repo.pp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+              }
+              default: {
+                fail { "Yum repository '${source}' is not supported for architecture ${hardwaremodel}": }
+              }
+            }
+            case $osrel {
+              '6': {
+                exec{ 'retrieve_clouderakey':
+                  command => "/usr/bin/curl -q http://archive.cloudera.com/cdh${cdhver}/redhat/6/${::hardwaremodel}/cdh/RPM-GPG-KEY-cloudera -o /etc/pki/rpm-gpg/RPM-GPG-KEY-cloudera",
+                  creates => '/etc/pki/rpm-gpg/RPM-GPG-KEY-cloudera',
```
Comment (deric) [inline manifests/repo.pp]: do we really need this, when GPG key URI is defined on line 43?

File: manifests/repo.pp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                file{ '/etc/pki/rpm-gpg/RPM-GPG-KEY-cloudera':
+                  mode => 0644,
+                  require => Exec["retrieve_clouderakey"],
+                }
+
+                yumrepo { "cloudera-cdh${cdhver}":
+                  ensure   => present,
+                  descr => "Cloudera's Distribution for Hadoop, Version ${cdhver}",
+                  baseurl => "http://archive.cloudera.com/cdh${cdhver}/redhat/6/${::hardwaremodel}/cdh/${cdhver}/",
+                  gpgkey => "http://archive.cloudera.com/cdh${cdhver}/redhat/6/${::hardwaremodel}/cdh/RPM-GPG-KEY-cloudera",
```
Comment (deric) [inline manifests/repo.pp]: either remove the exec above or `gpgkey` should be set to `/etc/pki/rpm-gpg/RPM-GPG-KEY-cloudera`, right? Or am I missing something?

**Comment by coreone** [comment]
Yeah, the GPG download wasn't necessary, so I removed it in my lastest commit.  Let me know if you have any other suggestions.

**Comment by deric** [comment]
Great, looks good. Thanks a lot!

**Pull request merged**